sqlglot/optimizer/qualify_columns.py

```diff
@@ -380,18 +410,89 @@ def _qualify_columns(scope: Scope, resolver: Resolver) -> None:
                     column.set("table", column_table)
 
 
+def _expand_struct_stars(
+    expression: exp.Dot,
+    next_alias_name: t.Callable[[], str],
+) -> t.List[exp.Alias]:
+    struct_fields: t.List[str] = []
+
+    # [BigQuery] Expand/Flatten foo.bar.* where bar is a struct column
+    starting_col = expression.find(exp.Column)
+    assert starting_col
+
+    dot_parts = expression.parts
+    table_name = starting_col.table
+
+    # If there's more nested levels specified, start the DFS traversal from the nested struct first
+    for part in dot_parts[2:-1]:
+        datatype = starting_col.find(exp.DataType) or (
+            starting_col.type and starting_col.type.find(exp.DataType)
+        )
```

Comment:
The second `find` is for the case where we are handling an `exp.Column` and is indeed superfluous, thanks!

The first find is because if we're handling `exp.ColumnDef` then the type is stored as `kind` arg, so `.type` does not work well here right?

Reply:
Can we just do this a bit more explicitly? This is not very obvious and I'm not sure I understand the implications of using `find` vs having a couple of branches to check the two cases, i.e. `Column` and `ColumnDef`.

If we can somehow handle `Column` before the loop to do make this cleaner, that'd be ideal.